src/BitStringAddresses/occupationnumberfs.jl

```diff
@@ -24,36 +26,68 @@ true
 
 julia> num_particles(ofs)
 6
+
+julia> OccupationNumberFS{5}() # vacuum state with 5 modes
+OccupationNumberFS{5, UInt8}(0, 0, 0, 0, 0)
+
+julia> OccupationNumberFS(i for i in 1:3) # use list comprehension
+OccupationNumberFS{3, UInt8}(1, 2, 3)
+
+julia> OccupationNumberFS(4, 1=>2, 3=>4) # sparse constructor
+OccupationNumberFS{4, UInt8}(2, 0, 4, 0)
 ```
 """
 struct OccupationNumberFS{M,T<:Unsigned} <: SingleComponentFockAddress{missing,M}
     onr::SVector{M,T}
+
+    function OccupationNumberFS{M,T}(args...) where {M,T<:Unsigned}
+        return new(SVector{M,T}(args...))
+    end
 end
 
-function OccupationNumberFS{M,T}(args...) where {M,T}
-    return OccupationNumberFS(SVector{M,T}(args...))
+function OccupationNumberFS(sv::SVector{M,T}) where {M,T<:Unsigned}
+    return OccupationNumberFS{M,T}(sv)
 end
 
-function OccupationNumberFS(args...)
-    sv = SVector(args...)
-    all(isinteger, sv) || throw(ArgumentError("all arguments must be integers"))
-    all(x -> x ≥ 0, sv) || throw(ArgumentError("all arguments must be non-negative"))
-    all(x -> x < 256, sv) || throw(ArgumentError("arguments don't fit in a byte, specify type"))
-    return OccupationNumberFS(SVector{length(sv),UInt8}(args...))
+function OccupationNumberFS(arg)
+    t = Tuple(arg)
+    return OccupationNumberFS{length(t)}(t)
 end
 
+function OccupationNumberFS(args::Integer...)
+    return OccupationNumberFS{length(args)}(args)
+end
+OccupationNumberFS(arg::Integer) = OccupationNumberFS{1}(arg) # to resolve ambiguity
+
 function OccupationNumberFS{M}(args...) where M
     sv = SVector{M}(args...)
```

Comment:
```suggestion
    sv = SVector{M}(args...)
    if all(x -> isa(x, Pair), sv)
        return OccupationNumberFS{M}(Tuple(sv))
    end
```

Is it worth adding something like this for consistency with `BoseFS` and `FermiFS`?

```
julia> OccupationNumberFS{5}(i=>i for i in 1:5)
ERROR: MethodError: no method matching isinteger(::Pair{Int64, Int64})

julia> BoseFS{15,5}(i=>i for i in 1:5)
BoseFS{15,5}(1, 2, 3, 4, 5)

julia> FermiFS{5,5}(i=>1 for i in 1:5)
FermiFS{5,5}(1, 1, 1, 1, 1)
```